src/libsyncengine/jobs/network/API_v2/uploadjob.cpp

```diff
@@ -184,8 +184,35 @@ std::string UploadJob::getContentType(bool &canceled) {
 }
 
 ExitInfo UploadJob::readFile() {
-    std::ifstream file(_filePath, std::ios_base::in | std::ios_base::binary);
-    if (!file.is_open()) {
+    std::ifstream file;
```

Comment:
Would it be possible to extract a method that would used twice?
Using the second form, we can inspect the `ExitInfo` output to decide whether the returned value is `true` or `false` in the first form.

Reply:
What about this modification? : https://github.com/Infomaniak/desktop-kDrive/pull/396/files/ffedf231d940bedb12aee00aff04d360fefeea8f..47d181ac98f136c35466863b15e7eb0e9d3a2a67

Reply:
I didn't move the comment in IoHelper::readFile as this is a generic function that I think could be used in the future in a context where this comment might be irrelevant.